File: src/main/java/dev/robothanzo/werewolf/model/SpeechOrder.java
package dev.robothanzo.werewolf.model;

import net.dv8tion.jda.api.entities.emoji.Emoji;

import java.util.Locale;

public enum SpeechOrder {
    UP, DOWN;

    public static SpeechOrder getRandomOrder() {
        return values()[(int) (Math.random() * values().length)];
    }

    public static SpeechOrder fromString(String s) {
        return valueOf(s.toUpperCase(Locale.ROOT));
    }

    @Override
    public String toString() {
        return this == UP ? "往上" : "往下";
    }

    public Emoji toEmoji() {
        return this == UP ? Emoji.fromUnicode("U+2b06") : Emoji.fromUnicode("U+2b07");
    }
}

File: src/main/kotlin/dev/robothanzo/werewolf/model/SpeechOrder.kt
package dev.robothanzo.werewolf.model

import net.dv8tion.jda.api.entities.emoji.Emoji
import java.util.*

enum class SpeechOrder {
    UP, DOWN;

    override fun toString(): String {
        return if (this == UP) "往上" else "往下"
    }

    fun toEmoji(): Emoji {
        return if (this == UP) Emoji.fromUnicode("U+2b06") else Emoji.fromUnicode("U+2b07")
    }

    companion object {
        fun getRandomOrder(): SpeechOrder {
            return entries[(Math.random() * entries.size).toInt()]
        }

        fun fromString(s: String): SpeechOrder {
            return valueOf(s.uppercase(Locale.ROOT))
        }
    }
}
